app/models/omniauth_data.rb

```diff
@@ -1,12 +1,48 @@
 class OmniauthData
 
+  VALID_PROVIDERS = ['identity', 'facebook', 'twitter', 'google_oauth2']
+
   def initialize(auth_data)
     @auth_data = auth_data
+    @info = @auth_data[:info] || {}
+    raise IllegalArgument unless VALID_PROVIDERS.include? provider
+  end
+
+  def provider
+    @auth_data[:provider]
+  end
+
+  def uid
+    @auth_data[:uid]
+  end
+
+  def name
+    @info[:name]
+  end
+
+  def nickname
+    # The User model will sanitize this
+    @info[:nickname] || name
+  end
+
+  def first_name
+    @info[:first_name]
+  end
+
+  def last_name
+    @info[:last_name]
+  end
+
+  def email
+    # Facebook only returns verified emails
+    # Twitter returns no emails
+    # Google can return unverified emails
```

Comment:
Yep. Line 93 in https://github.com/zquestz/omniauth-google-oauth2/blob/master/lib/omniauth/strategies/google_oauth2.rb